File: 0x08-python-more_classes/3-rectangle.py
#!/usr/bin/python3
"""
rectangle
"""


class Rectangle:
    """rectangle"""
    def ___init___(self, width=0, height=0):
        """rectangle"""
        self.height = height
        self.width = width

    @property
    def width(self):
        """width"""
        return self.__width

    @width.setter
    def width(self, value):
        """width"""
        if type(value) is not int:
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @property
    def height(self):
        """height"""
        return self.__height

    @height.setter
    def height(self, value):
        """width"""
        if type(value) is not int:
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    def area(self):
        """area"""
        return self.__width * self.__height

    def perimeter(self):
        """perimeter"""
        if self.__width == 0 or self.__height == 0:
            return 0
        return (self.__widtg * 2) + (self.__height * 2)

    def __str__(self):
        """string"""
        str = ""
        if self.__width != 0 and self.__height != 0:
            str += "\n".join("#" * self.__width
                             for j in range(self.__height))
        return str

File: 0x13-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
class Rectangle {
  constructor (w, h) {
    if ((w > 0) && (h > 0)) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    for (let n = 0; n < this.height; n++) {
      let space = '';
      for (let m = 0; m < this.width; m++) {
	space += 'X';
      }
      console.log(space);
    }
  }
}

module.exports = Rectangle;
